### PR TITLE
Small cleanups

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/shared/btrfs-util.c [hunk 1]
```diff
@@ -1637,13 +1637,10 @@ int btrfs_qgroup_find_parents(int fd, uint64_t qgroupid, uint64_t **ret) {
                         break;
         }
 
-        if (n_items <= 0) {
-                *ret = NULL;
-                return 0;
-        }
+        assert((n_items > 0) == (items != NULL));
+        assert(n_items <= INT_MAX);
```
Comment (YHNdnzj) [inline src/shared/btrfs-util.c:1641]: Is https://github.com/systemd/systemd/commit/5e30e6e28190fe30e76e071b2eb99546abcee7e6 no longer necessary with this?

**Review by YHNdnzj — APPROVED** [review]
LGTM, but one question.

File: src/shared/btrfs-util.c (hunk 1 above)
Comment (keszybz) [reply src/shared/btrfs-util.c:1641]: Possibly, but that's just a different check in a different place. I saw Luca's patch, and I think we can have both.

File: src/shared/btrfs-util.c [hunk 2]
```diff
@@ -1637,13 +1637,10 @@ int btrfs_qgroup_find_parents(int fd, uint64_t qgroupid, uint64_t **ret) {
                         break;
         }
 
-        if (n_items <= 0) {
-                *ret = NULL;
-                return 0;
-        }
+        assert((n_items > 0) == (items != NULL));
```
Comment (poettering) [inline src/shared/btrfs-util.c]: we usually don't explicit NULL comparisons?

hence turn rhs into `!!items`?

Comment (keszybz) [reply src/shared/btrfs-util.c]: Updated like that.

**Review by yuwata — APPROVED** [review]

**Pull request merged**